tokio/src/io/util/read_int.rs

```diff
@@ -0,0 +1,122 @@
+use crate::io::AsyncRead;
+
+use bytes::Buf;
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::io;
+use std::io::ErrorKind::UnexpectedEof;
+use std::mem::size_of;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+macro_rules! reader {
+    ($name:ident, $ty:ty, $reader:ident) => {
+        reader!($name, $ty, $reader, size_of::<$ty>());
+    };
+    ($name:ident, $ty:ty, $reader:ident, $bytes:expr) => {
+        pin_project! {
+            #[doc(hidden)]
```

Comment:
These `doc(hidden)` make me sad, since they mean the user can't even see that the return type is `impl Future`.

Reply:
I’m fine exposing more types, but this reminds me that I’ve been adding an “equivalent” async fn definition in docs.

Reply:
I'm hesitant to re-export __futures__ as, in theory, one day we will be able to use real `async fn` here.

That said, since it is 0.2, we can re-export them for now.